File: icpcD.py
for t in xrange(input()):
	n = input()
	a = map(int, raw_input().split())
	d = {}
	
	MAX = 75 * 10 ** 4
	M = 10 ** 9 + 7
	a.sort()
	for i in xrange(n):
		d[a[i]] = i

	i = n - 1
	ans = [0] * n
	while i >= 0:
		curr = a[i]
		ans[i] = 1
		for j in xrange(curr * 2, a[n - 1] + 1, curr):
			if j in d:
				ans[i] = (ans[d[j]] + ans[i]) % M
		i -= 1
	# print ans
	# print d
	fAns = 0
	for i in ans:
		fAns = (fAns + i) % M
	print fAns	

File: icpcD.cpp
#include <bits/stdc++.h>
using namespace std;
#define M 1000000007
int t, n, a[751234];

int main() {
	cin >> t;

	while (t--) {
		cin >> n;

		for (int i = 0; i < n; i++) {
			cin >> a[i];
		}

		sort(a, a + n);

		unordered_map <int, int> m;
		for (int i = 0; i < n; i++) {
			m[a[i]] = i;
		}

		int i = n - 1;
		int ans[n];
		while (i >= 0) {
			int curr = a[i];
			ans[i] = 1;
			for (int j = curr * 2; j <= 751234; j += curr) {
				if (m.find(j) != m.end()) {
					ans[i] = (ans[i] + ans[j]) % M;
				}
			}
			i -= 1;
		}

		int fAns = 0;
		for (int i = 0; i < n; i++) {
			fAns = (fAns + ans[i]) % M;
		}

		cout << fAns << endl;
	}
}